src/Application/Messenger/DonationUpdated.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace MatchBot\Application\Messenger;
+
+use MatchBot\Application\Assertion;
+use MatchBot\Domain\Donation;
+
+class DonationUpdated extends AbstractStateChanged
+{
+    private function __construct(
+        public string $uuid,
+        public ?string $salesforceId,
+        public array $json,
+    ) {
+        // TODO Things like Stripe hooks could come back before pushing to Salesforce is complete, so we need to be
```

Comment:
Realised late due to integration tests that asserting this isn't blank is probably a design flaw, because Stripe hooks might come in before SF ID is set. I guess one option is to deliberately error and HTTP 500 on those so that Stripe resend but that feels unnecessarily brittle. We might need to think about bring back a recovery mechanism where we do talk to the database if we get an update job with no SF ID. Or possibly pause for a few seconds and re-queue it?

Reply:
I'm also not sure if you expected the id to always be not null why you would use `Assertion::notNull($this->salesforceId);` rather than typing it as `string` instead of `?string`